src/Miki/Modules/GuildAccounts/GuildAccountsModule.cs

```diff
@@ -26,93 +22,48 @@ public class GuildAccountsModule
-            var localExperienceRepository = unit.GetRepository<LocalExperience>();
-            var timerRepository = unit.GetRepository<Timer>();
-
-            LocalExperience thisUser = await localExperienceRepository.GetAsync(
-                (long)e.GetGuild().Id, (long)e.GetAuthor().Id);
-            // TODO move to LocalExperience service?
-            if(thisUser == null)
-            {
-                thisUser = new LocalExperience
-                {
-                    ServerId = (long)e.GetGuild().Id,
-                    UserId = (long)e.GetAuthor().Id,
-                    Experience = 0,
-                };
-                await localExperienceRepository.AddAsync(thisUser)
-                    .ConfigureAwait(false);
-                await unit.CommitAsync().ConfigureAwait(false);
-            }
-
-            GuildUser thisGuild = await guildService.GetGuildAsync((long)e.GetGuild().Id);
-
-            if(thisUser.Experience < thisGuild.MinimalExperienceToGetRewards)
-            {
-                await e.ErrorEmbedResource("miki_guildweekly_insufficient_exp",
-                        thisGuild.MinimalExperienceToGetRewards.ToString("N0"))
-                    .ToEmbed()
-                    .QueueAsync(e, e.GetChannel());
-                return;
-            }
-                
-            Timer timer = await timerRepository.GetAsync((long)e.GetGuild().Id, (long)e.GetAuthor().Id);
-            if (timer == null)
-            {
-                timer = await timerRepository.AddAsync(
-                    new Timer
-                {
-                    GuildId = (long)e.GetGuild().Id,
-                    UserId = (long)e.GetAuthor().Id,
-                    Value = DateTime.Now.AddDays(-30)
-                });
-                await unit.CommitAsync();
-            }
+            var response = await guildService.ClaimWeeklyAsync(
+                new GuildUserReference((long)e.GetGuild().Id, (long)e.GetAuthor().Id));
 
-            if (timer.Value.AddDays(7) > DateTime.Now)
-            {
-                await e.ErrorEmbedResource(
-                        "guildweekly_error_timer_running", 
-                        (timer.Value.AddDays(7) - DateTime.Now).ToTimeString(e.GetLocale()))
-                    .ToEmbed()
-                    .QueueAsync(e, e.GetChannel());
-                return;
-            }
+            var embed = new EmbedBuilder()
+                .SetTitle(":house: " + locale.GetString("guildweekly_title"))
+                .SetColor(255, 232, 182);
 
-            GuildUser rival = await guildService.GetRivalAsync(thisGuild);
-            if (rival.Experience > thisGuild.Experience)
+            switch (response.Status)
             {
-                await e.ErrorEmbedResource("guildweekly_error_low_level")
-                    .ToEmbed()
-                    .QueueAsync(e, e.GetChannel());
-                return;
+                case WeeklyStatus.Success:
```

Comment:
Would be cool to have this as a IResource to make the command logic smaller :)

Reply:
Check this commit for an example https://github.com/Mikibot/bot/commit/c11576418a758a2b57108bb5681deef3162dae1b